databricks_cli/clusters/cli.py

```diff
@@ -0,0 +1,188 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+from tabulate import tabulate
+
+from databricks_cli.click_types import OutputClickType
+from databricks_cli.clusters.api import create_cluster, start_cluster, restart_cluster, \
+    delete_cluster, get_cluster, list_clusters, list_zones, list_node_types, spark_versions
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, pretty_format, json_cli_base
+from databricks_cli.configure.config import require_config
+from databricks_cli.version import print_version_callback
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None,
+              help='File containing json to POST to /clusters/create.')
+@click.option('--json', default=None, help='Displays absolute paths.')
+@require_config
+@eat_exceptions
+def create_cli(json_file, json):
+    """
+    Creates a Databricks Cluster
+
+    The specification for the request json can be found at
+    https://docs.databricks.com/api/latest/clusters.html#create
+    """
+    json_cli_base(json_file, json, create_cluster)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--cluster-id', required=True)
+@require_config
+@eat_exceptions
+def start_cli(cluster_id):
+    """
+    Starts a terminated Spark cluster given its id.
+
+    If the cluster is not currently in a TERMINATED state, nothing will happen.
+
+    """
+    start_cluster(cluster_id)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--cluster-id', required=True)
+@require_config
+@eat_exceptions
+def restart_cli(cluster_id):
+    """
+    Restarts a Spark cluster given its id.
+
+    If the cluster is not currently in a RUNNING state, nothing will happen
+    """
+    restart_cluster(cluster_id)
```

Comment:
does it return different response to indicate success/failure?

Reply:
Yup it will use exitcode 1. This function does the magic

```
 36 def eat_exceptions(function):
 37     @six.wraps(function)
 38     def decorator(*args, **kwargs):
 39         try:
 40             return function(*args, **kwargs)
 41         except HTTPError as exception:
 42             if exception.response.status_code == 401:
 43                 error_and_quit('Your authentication information may be incorrect. Please '
 44                                'reconfigure with ``dbfs configure``')
 45             else:
 46                 error_and_quit(exception.response.content)
 47         except Exception as exception: # noqa
 48             if not DEBUG_MODE:
 49                 error_and_quit('{}: {}'.format(type(exception).__name__, str(exception)))
 50     decorator.__doc__ = function.__doc__
 51     return decorator```

Reply:
I guess I mean how user could tell if the cluster is restarted or not. It seems the api doesn't tell you that.